interop-testing/src/main/java/io/grpc/testing/integration/Util.java

```diff
@@ -66,4 +71,32 @@ public static void assertEquals(List<? extends MessageLite> expected,
       }
     }
   }
+
+  static List<SocketAddress> getV6Addresses(int port) throws UnknownHostException {
+    List<SocketAddress> v6addresses = new ArrayList<>();
+    InetAddress[] addresses = InetAddress.getAllByName(InetAddress.getLocalHost().getHostName());
+    for (InetAddress address : addresses) {
+      if (address.getAddress().length != 4) {
+        v6addresses.add(new java.net.InetSocketAddress(address, port));
+      }
+    }
+    return v6addresses;
+  }
+
+  static SocketAddress getV4Address(int port) throws UnknownHostException {
+    InetAddress[] addresses = InetAddress.getAllByName(InetAddress.getLocalHost().getHostName());
+    for (InetAddress address : addresses) {
+      if (address.getAddress().length == 4) {
+        return new java.net.InetSocketAddress(address, port);
+      }
+    }
+    return null; // should never happen
```

Comment:
This can totally 100% happen. In fact, it is happening with the Mac machine now.